src/bin/snx-rs.rs

```diff
@@ -140,3 +146,35 @@ async fn main() -> anyhow::Result<()> {
 
     result
 }
+
+async fn get_server_info(params: &TunnelParams) -> anyhow::Result<ServerInfoResponse> {
+    let client = CccHttpClient::new(Arc::new(params.clone()), None);
+    let info = client.get_server_info().await?;
+    let response_data = info.get("ResponseData").unwrap_or(&Value::Null);
+    Ok(serde_json::from_value::<ServerInfoResponse>(response_data.clone())?)
+}
+
+async fn get_server_pwd_prompts(params: &TunnelParams) -> anyhow::Result<VecDeque<String>> {
```

Comment:
Looks like a duplicated code from the controller.rs, I would extract it somewhere.

Reply:
Moved it to server_info module of platform.